api/src/main/java/ca/bc/gov/educ/api/distribution/process/BaseProcess.java

```diff
@@ -82,17 +82,22 @@ protected void createZipFile(Long batchId, ProcessorData processorData) {
         logger.debug("Create zip file for {}", processorData.getActivityCode());
         StringBuilder sourceFileBuilder = new StringBuilder().append(EducDistributionApiConstants.TMP_DIR).append(EducDistributionApiConstants.DEL).append(batchId);
         File file = new File(EducDistributionApiConstants.TMP_DIR + "/EDGRAD.BATCH." + batchId + ".zip");
-        writeZipFile(sourceFileBuilder, file);
+        writeZipFile(sourceFileBuilder.toString(), file);
     }
 
-    protected void writeZipFile(StringBuilder sourceFileBuilder, File file) {
-        try (FileOutputStream fos = new FileOutputStream(file)) {
+    protected void writeZipFile(String rootPath, File file) {
+        try (FileOutputStream fos = new FileOutputStream(file.getAbsolutePath())) {
             ZipOutputStream zipOut = new ZipOutputStream(fos);
-            File fileToZip = new File(sourceFileBuilder.toString());
+            File fileToZip = new File(rootPath);
             EducDistributionApiUtils.zipFile(fileToZip, fileToZip.getName(), zipOut);
+            zipOut.close();
         } catch (IOException e) {
             logger.debug(EXCEPTION, e.getLocalizedMessage());
         }
+        /**
+        if(!EducDistributionApiUtils.isValid(file)) {
+            throw new GradBusinessRuleException("Zip file " + file.getAbsolutePath() + " is not valid");
+        }**/
     }
```

Comment:
Why are we not checking if valid now?

Reply:
We never did it before. I used this code pure for debugging and see which file is bad.

Reply:
May be we should have the check... Can you ask KIm that the whole process will be set as failed if there are bad files in the zip? May be the better just have a process to regenerate corrupted files?

Reply:
Oh ok. Yes I think this would be a good check, but you can bring it up at the scrum.